tests/testthat/test-index.R

```diff
@@ -0,0 +1,51 @@
+context("customindex")
+
+test_that("Omitting required template keys produces warnings", {
+  string <-
+    "<!DOCTYPE html>
+  <html>
+  <head>
+  {%meta_tags%}
+  <title>Testing Again</title>
+  {%favicon%}
+  {%css_tags%}
+  </head>
+  <body>
+  {%app_entry%}
+  <footer>
+  {%config%}
+  {%scripts%}
+  </footer>
+  </body>
+  </html>"
+  
+  app <- Dash$new()
```

Comment:
I'm assuming meta_tags, favicon, css_tag are optional? Might as well test that too for completeness.

Reply:
Indeed, they are optional, but can test them easily enough, will add.

Reply:
Actually, those go in the integration test, since they're tested with Selenium. Out of scope for this PR, but we can consider them for the `pytest` script.